File: friendly_cars_frontend/src/service/ShoppingCartService.js
import axios from "axios";

const SHOPPING_CART_BASE_URL = "http://localhost:8080/api/v1/carts";

class ShoppingCartService {
    createShoppingCart(clientId, shoppingCart) {
        return axios.post(SHOPPING_CART_BASE_URL + `/c/${clientId}`, shoppingCart)
            .then(response => response.data)
    }

    getAllShoppingCarts() {
        return axios.get(SHOPPING_CART_BASE_URL)
            .then(response => response.data)
    }

    getShoppingCartById(shoppingCartId) {
        return axios.get(SHOPPING_CART_BASE_URL + `/${shoppingCartId}`)
            .then(response => response.data)
    }

    getShoppingCartByClientId(clientId) {
        return axios.get(SHOPPING_CART_BASE_URL + `/c/${clientId}`)
            .then(response => response.data)
    }

    updateShoppingCart(shoppingCartId, newShoppingCart) {
        return axios.put(SHOPPING_CART_BASE_URL + `/${shoppingCartId}`, newShoppingCart)
            .then(response => response.data)
    }

    partialUpdateShoppingCart(shoppingCartId, newShoppingCart) {
        return axios.patch(SHOPPING_CART_BASE_URL + `/${shoppingCartId}`, newShoppingCart)
            .then(response => response.data)
    }

    deleteShoppingCart(shoppingCartId) {
        return axios.delete(SHOPPING_CART_BASE_URL + `/${shoppingCartId}`)
            .then(response => response.data)
    }

    dropVehicleOfShoppingCart(shoppingCartId, vehicleId) {
        return axios.post(SHOPPING_CART_BASE_URL + `/${shoppingCartId}/vh/${vehicleId}`)
            .then(response => response.data)
    }
}

export default new ShoppingCartService()

File: cs_manager_system/src/main/java/com/csManager/csmanager/service/ShoppingCartService.java
package com.csManager.csmanager.service;

import java.util.List;
import java.util.Map;

import com.csManager.csmanager.entity.ShoppingCart;
import com.csManager.csmanager.exception.ClientNotFoundException;
import com.csManager.csmanager.exception.ShoppingCartNotFoundException;

import org.springframework.http.ResponseEntity;

/**
 * ShoppingCartService
 */
public interface ShoppingCartService {
    ResponseEntity<List<ShoppingCart>> getAllShoppingCarts();
    ResponseEntity<ShoppingCart> getShoppingCartById(long shoppingCartId)
            throws ShoppingCartNotFoundException;
    ResponseEntity<ShoppingCart> getShoppingCartClientId(String clientId)
            throws ClientNotFoundException;
    ResponseEntity<ShoppingCart> updateShoppingCart(long shoppingCartId, ShoppingCart updatedShoppingCart)
            throws ShoppingCartNotFoundException;
    ResponseEntity<ShoppingCart> partialUpdateShoppingCart(long shoppingCartId, Map<Object, Object> fields)
            throws ShoppingCartNotFoundException;
    ResponseEntity<?> deleteShoppingCart(long shoppingCartId)
            throws ShoppingCartNotFoundException;
    ResponseEntity<ShoppingCart> createShoppingCart(String clientId, ShoppingCart shoppingCart)
            throws ClientNotFoundException;
    ResponseEntity<ShoppingCart> dropVehicleOfShoppingCart(long shoppingCartId, long vehicleId)
            throws Exception;
}
